### PR TITLE
Events detectors

### PR DESCRIPTION
**Pull request by sofiazcoaga**
Includes: 

- Storage change events detector: emits a warning when storage is modified in any way and the contract does not emit an event. 
- Storage change events test cases. 
- Token Interface events detector: emits a warning when the token interface trait is implemented and the functions belonging to it that should emit events do not. 
- Token Interface events test cases.

### CODE REVIEW

File: test-cases/token-interface-events/token-interface-events-1/vulnerable-example/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        Self::get_metadata(env.clone()).admin.require_auth();
+        let previous_balance: i128 = env
+            .clone()
+            .storage()
+            .instance()
+            .get(&DataKey::Balance(to.clone()))
+            .unwrap_or(0);
+        env.storage()
+            .instance()
+            .set(&DataKey::Balance(to), &(previous_balance + amount));
```
Comment (tenuki) [inline test-cases/token-interface-events/token-interface-events-1/vulnerable-example/src/lib.rs:89]: Esta función está modificando balances/ y no tira eventos.. y el detector de token-interface no dice nada y deberia, no?

Comment (tenuki) [reply test-cases/token-interface-events/token-interface-events-1/vulnerable-example/src/lib.rs:89]: ver que es un evento definido en la TI : https://github.com/stellar/rs-soroban-sdk/blob/f57d2454e520ae11dc9c8d55db610557a24a556e/soroban-token-sdk/src/event.rs#L25C18-L25C22

**Review by tenuki — CHANGES_REQUESTED** [review]

**Comment by tenuki** [comment]
sofi check the comments let us know whatever you thinkg about it..

File: test-cases/token-interface-events/token-interface-events-1/vulnerable-example/src/lib.rs (hunk 1 above)
Comment (sofiazcoaga) [reply test-cases/token-interface-events/token-interface-events-1/vulnerable-example/src/lib.rs:89]: hecho

**Review by tenuki — APPROVED** [review]

**Pull request merged**